lifecycle.go

```diff
@@ -26,6 +26,11 @@ import (
 	"go.uber.org/fx/internal/lifecycle"
 )
 
+// A HookFunc is a function that can be used as a [Hook].
+type HookFunc interface {
+	~func() | ~func() error | ~func(context.Context) | ~func(context.Context) error
+}
```

Comment:
Since according to the comment in `internal/lifecycle/lifecycle.go`, `lifecycle.Callable` should be identical to `fx.HookFunc`, is there a particular reason why this can't be just a type alias to `lifecycle.Callable`? Will it break documentation?

Reply:
It can break docs, but more generally, I don't want to create a type alias for something that isn't itself exported. I don't think keeping them in sync here will require any effort.